### PR TITLE
test: android test enhancements

### PR DESCRIPTION
**Pull request by mxschmitt**
Extracted from #15777.

### CODE REVIEW

File: tests/page/locator-highlight.spec.ts [hunk 1]
```diff
@@ -16,16 +16,34 @@
...
 });
+
+function roundBox(box: BoundingBox): BoundingBox {
+  return {
+    x: Math.round(box.x),
+    y: Math.round(box.y),
+    width: Math.round(box.width),
+    height: Math.round(box.height),
+  };
+}
```
Comment (rwoll) [inline tests/page/locator-highlight.spec.ts:49]: Will this deflake it? Might be better to check they are mostly overlapping instead. `overlapping(box1, box2, tolerance)`

File: utils/avd_stop.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+set -e
+
+if [[ -z "${ANDROID_HOME}" ]]; then
+    SDKDIR=$PWD/.android-sdk
+    export ANDROID_HOME=${SDKDIR}
+    export ANDROID_SDK_ROOT=${SDKDIR}
+fi
+
+echo "Killing previous emulators"
+${ANDROID_HOME}/platform-tools/adb devices | grep emulator | cut -f1 | while read line; do ${ANDROID_HOME}/platform-tools/adb -s $line emu kill; done
```
Comment (rwoll) [inline utils/avd_stop.sh:12]: Why is this a separate file? Looks only used on one place and makes it hard to follow `utils/avd_start.sh`?

**Review by rwoll — APPROVED** [review]

File: tests/page/locator-highlight.spec.ts (hunk 1 above)
Comment (mxschmitt) [reply tests/page/locator-highlight.spec.ts:49]: yes, it will deflate, its 0.00001 off on Android.

File: utils/avd_stop.sh (hunk 2 above)
Comment (mxschmitt) [reply utils/avd_stop.sh:12]: It's helpful when running avd locally.
